ext/pdo/php_pdo_driver.h

```diff
@@ -432,42 +432,47 @@ struct _pdo_dbh_t {
 	void *driver_data;
 
 	/* credentials */
-	char *username, *password;
+	char *username;
+	char *password;
 
 	/* if true, then data stored and pointed at by this handle must all be
 	 * persistently allocated */
-	unsigned is_persistent:1;
+	uint8_t is_persistent:1;
 
 	/* if true, driver should act as though a COMMIT were executed between
 	 * each executed statement; otherwise, COMMIT must be carried out manually
 	 * */
-	unsigned auto_commit:1;
+	uint8_t auto_commit:1;
 
 	/* if true, the handle has been closed and will not function anymore */
-	unsigned is_closed:1;
+	uint8_t is_closed:1;
 
 	/* if true, the driver requires that memory be allocated explicitly for
 	 * the columns that are returned */
-	unsigned alloc_own_columns:1;
+	uint8_t alloc_own_columns:1;
 
 	/* if true, commit or rollBack is allowed to be called */
-	bool in_txn:1;
+	uint8_t in_txn:1;
 
-	/* max length a single character can become after correct quoting */
-	unsigned max_escaped_char_length:3;
+	/* when set, convert int/floats to strings */
+	uint8_t stringify:1;
 
 	/* oracle compat; see enum pdo_null_handling */
-	unsigned oracle_nulls:2;
+	uint8_t oracle_nulls:2;
 
-	/* when set, convert int/floats to strings */
-	unsigned stringify:1;
+	/* The PDO Error mode pdo_error_mode */
+	uint8_t error_mode:2;
 
-	/* bitmap for pdo_param_event(s) to skip in dispatch_param_event */
-	unsigned skip_param_evt:7;
+	/* max length a single character can become after correct quoting */
+	uint8_t max_escaped_char_length:3;
+
+	/* Case conversion pdo_case_conversion */
+	uint8_t native_case:2;
+	uint8_t desired_case:2;
+	uint8_t _reserved:4;
 
-	/* the sum of the number of bits here and the bit fields preceding should
-	 * equal 32 */
-	unsigned _reserved_flags:14;
+	/* bitmap for pdo_param_event(s) to skip in dispatch_param_event */
+	uint8_t skip_param_evt;
```

Comment:
Annoyingly, being able to specify the size of a C enum is a C23 feature, otherwise we could just use `pdo_param_event` here.